### PR TITLE
[chip-tool] Add busyWaitMs optional argument to chip-tool to lock the…

### PR DESCRIPTION
**Pull request by vivien-apple**
… main thread and the reception of messages for a given duration once a message is sent

#### Problem

The test suite uses a hack to lock the main thread for making it possible to test timed interaction. This can not be reproduce in `chip-tool`.

### CODE REVIEW

File: src/app/tests/suites/commands/interaction_model/InteractionModel.h [hunk 1]
```diff
@@ -155,6 +155,7 @@ class InteractionModelCommands
     CHIP_ERROR SendCommand(chip::DeviceProxy * device, chip::EndpointId endpointId, chip::ClusterId clusterId,
                            chip::CommandId commandId, const T & value,
                            const chip::Optional<uint16_t> & timedInteractionTimeoutMs = chip::NullOptional,
+                           const chip::Optional<uint16_t> & busyWaitForMs             = chip::NullOptional,
```
Comment (andy31415) [inline src/app/tests/suites/commands/interaction_model/InteractionModel.h:158]: we changed a command from 9 arguments to now take 10 arguments. 4 of those are optional<uint16_t>. This is probably not long term maintainable ..... any chance we can use some builder pattern and deprecate this particular method?

File: src/app/tests/suites/commands/interaction_model/InteractionModel.h [hunk 2]
```diff
@@ -264,14 +288,15 @@ class InteractionModelWriter
     CHIP_ERROR WriteAttribute(chip::DeviceProxy * device, std::vector<chip::EndpointId> endpointIds,
                               std::vector<chip::ClusterId> clusterIds, std::vector<chip::AttributeId> attributeIds, const T & value,
                               const chip::Optional<uint16_t> & timedInteractionTimeoutMs          = chip::NullOptional,
+                              const chip::Optional<uint16_t> & busyWaitForMs                      = chip::NullOptional,
```
Comment (andy31415) [inline src/app/tests/suites/commands/interaction_model/InteractionModel.h:291]: and this one now has 11 arguments. ... same complaint: this is not long term maintanable and invocations of this API is not code reviewable for correctness.

**Review by bzbarsky-apple — APPROVED** [review]

File: src/app/tests/suites/commands/interaction_model/InteractionModel.h (hunk 2 above)
Comment (vivien-apple) [reply src/app/tests/suites/commands/interaction_model/InteractionModel.h:291]: @andy31415  I made https://github.com/project-chip/connectedhomeip/pull/24091 to address this issue. I will rebased this one on ToT once https://github.com/project-chip/connectedhomeip/pull/24091 has landed.

File: src/app/tests/suites/commands/interaction_model/InteractionModel.h (hunk 1 above)
Comment (vivien-apple) [reply src/app/tests/suites/commands/interaction_model/InteractionModel.h:158]: @andy31415  I made https://github.com/project-chip/connectedhomeip/pull/24091 to address this issue. I will rebased this one on ToT once https://github.com/project-chip/connectedhomeip/pull/24091 has landed.

File: src/app/tests/suites/commands/interaction_model/InteractionModel.h (hunk 2 above)
Comment (vivien-apple) [reply src/app/tests/suites/commands/interaction_model/InteractionModel.h]: I have rebased this patch on top of #24091 now that it has landed.

**Comment by andy31415** [comment]
@vivien-apple - do we need this after #24063 ?

**Comment by bzbarsky-apple** [comment]
@andy31415 Yes, see the PR description in https://github.com/project-chip/connectedhomeip/pull/24063, which explicitly talks about how it differs from this PR.  They address different use cases.

**Review by andy31415 — APPROVED** [review]

**Pull request merged**